src/utils/hooks/api/use-get-vault-overview.tsx

```diff
@@ -1,80 +1,73 @@
-  accountId: AccountId
-): Promise<VaultOverview> => {
-  const tokenIdLiteral = tickerToCurrencyIdLiteral(vault.backingCollateral.currency.ticker) as CollateralIdLiteral;
-
-  const apy = await window.bridge.vaults.getAPY(accountId, tokenIdLiteral);
-  const collateralization = await window.bridge.vaults.getVaultCollateralization(accountId, tokenIdLiteral);
+type VaultTotals = {
+  totalLockedCollateral: number;
+  totalUsdRewards: number;
+  totalAtRisk: number;
+};
 
-  const issues = await fetch(HYDRA_URL, {
-    method: 'POST',
-    headers: {
-      'Content-Type': 'application/json'
-    },
-    body: JSON.stringify({
-      query: issueCountQuery(`vault: {accountId_eq: "${accountId.toString()}"}, status_eq: Pending`)
-    })
-  });
+interface VaultOverview {
+  vaults: Array<VaultData> | undefined;
+  totals: VaultTotals | undefined;
+}
 
-  const issuesCount = await issues.json();
+const getVaultTotals = (vaults: Array<VaultData>) => ({
+  totalLockedCollateral: vaults.reduce((a, b) => a + b.collateral.usd, 0),
+  totalUsdRewards: vaults.reduce((a, b) => a + b.governanceTokenRewards.usd + b.wrappedTokenRewards.usd, 0),
+  totalAtRisk: vaults.map((vault) => vault.vaultAtRisk).filter(Boolean).length
+});
 
-  return {
-    apy,
-    collateralization,
-    issues: issuesCount.data.issuesConnection.totalCount,
-    collateralId: tokenIdLiteral,
-    wrappedId: VAULT_WRAPPED
-  };
-};
+const useGetVaultOverview = ({ address }: { address: string }): VaultOverview | undefined => {
+  const [queriesComplete, setQueriesComplete] = useState<boolean>(false);
+  const [queryError, setQueryError] = useState<Error | undefined>(undefined);
 
-const useGetVaultOverview = ({ address }: { address: string; }): Array<VaultOverview> => {
-  // TODO: can we handle this check at the application level rather than in components and utilties?
-  // https://www.notion.so/interlay/Handle-api-loaded-check-at-application-level-38fe5d146c8143a88cef2dde7b0e19d8
-  const { bridgeLoaded } = useSelector((state: StoreType) => state.general);
+  const { prices } = useSelector((state: StoreType) => state.general);
   const vaults = useGetVaults({ address });
+  useErrorHandler(queryError);
 
-  // TODO: updating react-query to > 3.28.0 will allow us to type this properly
-  const vaultData: Array<any> = useQueries<Array<UseQueryResult<unknown, unknown>>>(
-    vaults.map(vault => {
+  // TODO: updating react-query to > 3.28.0 will allow us type this without `any`
+  const vaultData: Array<any> = useQueries<Array<UseQueryResult<VaultOverview, Error>>>(
+    vaults.map((vault) => {
       return {
         queryKey: ['vaultsOverview', address, vault.backingCollateral.currency.ticker],
-        queryFn: () => getVaultOverview(vault, newAccountId(window.bridge.api, address)),
+        queryFn: () => getVaultOverview(vault, newAccountId(window.bridge.api, address), prices),
         options: {
-          enabled: !!bridgeLoaded
+          enabled: !!vault
```

Comment:
Why do we need this check? Can the vault ever be nullish? :thinking:

Reply:
Hmm, good point—it was there originally because it could be, but I refactored useGetVaults to only return once all queries had completed, so probably not. I'll take it out.